### ISSUE TITLE
[Suggestion]Compact workspace switcher applet

### ISSUE THREAD

**Issue by flavienbonvin** [budgie-desktop/budgie-desktop#1139]
Could be great to have a compact workspace switcher applet that display the number of the screen you are in instead of all the worspaces displayed in the row and/or a grid version of your workspaces (like kde do with pager)
@cybre

**Comment by dgaw** [comment]
I agree that the workspace switcher could benefit from a more compact UI. 

@flavienbonvin Have a look at an [alternative switcher](https://github.com/dgaw/budgie-workspaces-compact) I developed. I think it's in line with your suggestion.

**Comment by Jacob-Vlijm** [comment]
What version did you develop it on? doesn' t appear in the list, not after a budgie-panel --replace & (Ubuntu Budgie 17.10, beta1/10.4).

**Comment by dgaw** [comment]
@Jacob-Vlijm  I'm on Solus 3 with the packaged budgie-desktop (10.4, release: 146). Did you get any errors when running install.sh or budgie-panel --replace? If so, please paste.

**Comment by Jacob-Vlijm** [comment]
@dgaw Wait, got it! overlooked it, since the icon is exactly the same as the other worspace applet.

**Comment by Jacob-Vlijm** [comment]
@dgaw  It immediately made my panel freeze though.

**Comment by chifreagle** [comment]
working as expected on solus 3 (updated)

**Comment by Jacob-Vlijm** [comment]
second attempt also broke, third runs so far. I like the circular approach :)

**Comment by dgaw** [comment]
@Jacob-Vlijm Regarding the freeze, it would be great if you could paste any console errors/warnings (if you get any).

**Comment by dgaw** [comment]
@chifreagle Thanks for confirming.

**Comment by Jacob-Vlijm** [comment]
@dgaw  Will keep an eye on the terminal.

**Comment by Jacob-Vlijm** [comment]
@dgaw not a single issue afterwards. seems the panel did catch a cold or something.

**Comment by dgaw** [comment]
@Jacob-Vlijm Great. Cheers!

**Comment by chifreagle** [comment]
I think a new applet is just a temporary solution, we need a compact mode option in the original applet setting

**Comment by Jacob-Vlijm** [comment]
@chifreagle I am not sure I understand what you mean; what you *describe* ("applet that display the number of the screen you are in instead of all the worspaces displayed in the row") is pretty much a number in the panel, corresponding to the current workspace. What you *mention as an example* (kde pager) is quite different though.

**Comment by dgaw** [comment]
@chifreagle I agree about getting this into the original applet. For the time being though a separate applet is easier for me to prototype.

**Comment by flavienbonvin** [comment]
@dgaw that's exactly it! It has to be in the official applet tho

**Comment by dgaw** [comment]
Just a quick update on the [alternative switcher](https://github.com/dgaw/budgie-workspaces-compact) . I've added a right-click menu where you can set the number of workspaces.

The updated version is on github so you can download/reinstall if that's something of use.
